src/AuthorizeDotNet.php

```diff
@@ -0,0 +1,301 @@
+
+        $this->oAuthentication = new AuthNetAPI\MerchantAuthenticationType();
+        $this->oAuthentication->setName($this->getSetting('sLoginId'));
+        $this->oAuthentication->setTransactionKey($this->getSetting('sTransactionKey'));
+    }
+
+    // --------------------------------------------------------------------------
+
+    public function getApiMode()
+    {
+        return $this->sApiMode;
+    }
+
+    // --------------------------------------------------------------------------
+
+    public function getAuthentication()
+    {
+        return clone $this->oAuthentication;
+    }
+
+    // --------------------------------------------------------------------------
+
+    /**
+     * Returns whether the driver is available to be used against the selected invoice
+     * @return boolean
+     */
+    public function isAvailable($oInvoice)
+    {
+        return true;
+    }
+
+    // --------------------------------------------------------------------------
+
+    /**
+     * Returns whether the driver uses a redirect payment flow or not.
+     * @return boolean
+     */
+    public function isRedirect()
+    {
+        return false;
+    }
+
+    // --------------------------------------------------------------------------
+
+    /**
+     * Returns the payment fields the driver requires, use CARD for basic credit
+     * card details.
+     * @return mixed
+     */
+    public function getPaymentFields()
+    {
+        return 'CARD';
+    }
+
+    // --------------------------------------------------------------------------
+
+    /**
+     * Initiate a payment
+     *
+     * @param  integer   $iAmount      The payment amount
+     * @param  string    $sCurrency    The payment currency
+     * @param  \stdClass $oData        The driver data object
+     * @param  \stdClass $oCustomData  The custom data object
+     * @param  string    $sDescription The charge description
+     * @param  \stdClass $oPayment     The payment object
+     * @param  \stdClass $oInvoice     The invoice object
+     * @param  string    $sSuccessUrl  The URL to go to after successful payment
+     * @param  string    $sFailUrl     The URL to go to after failed payment
+     * @param  string    $sContinueUrl The URL to go to after payment is completed
+     *
+     * @return \Nails\Invoice\Model\ChargeResponse
+     */
+    public function charge(
+        $iAmount,
+        $sCurrency,
+        $oData,
+        $oCustomData,
+        $sDescription,
+        $oPayment,
+        $oInvoice,
+        $sSuccessUrl,
+        $sFailUrl,
+        $sContinueUrl
+    ) {
+
+        $oChargeResponse = Factory::factory('ChargeResponse', 'nailsapp/module-invoice');
+
+        try {
+
+            if (!empty($oInvoice->customer->billing_email)) {
+                $sReceiptEmail = $oInvoice->customer->billing_email;
+            } else {
+                $sReceiptEmail = $oInvoice->customer->email;
+            }
```

Comment:
`$sReceiptEmail` is not used, remove? Or can it be passed to AuthDotNet as meta data?